### PR TITLE
add EventInteractor

### PR DESCRIPTION
**Pull request by yshrsmz**
## TODO

- [x] unit tests

### CODE REVIEW

File: bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventInteractor.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    updateEventsAndNotify()
+  }
+
+  fun trackFetchEvaluationsFailure(
+    featureTag: String,
+    error: BKTException
+  ) {
+    val event = when (error) {
+      is BKTException.NetworkException,
+      is BKTException.TimeoutException -> newTimeoutErrorCountMetricsEvent(
```
Comment (yshrsmz) [inline bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventInteractor.kt:78]: currently mapping both NetworkException and TimeoutException to `TimeoutErrorCountMetricsEvent`

Comment (cre8ivejp) [reply bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventInteractor.kt:78]: I'm thinking of adding a `NetworkErrorCountMetricsEvent` to report it separately.
I will confirm and return to you next week, but for now, it is okay.

File: bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventInteractor.kt [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  private val clock: Clock,
+  private val idGenerator: IdGenerator,
+) {
+
+  @VisibleForTesting
+  internal val events: AtomicReference<List<Event>> = AtomicReference(emptyList())
+
+  private var eventUpdateListener: EventUpdateListener? = null
+
+  fun setEventUpdateListener(listener: EventUpdateListener?) {
```
Comment (cre8ivejp) [inline bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventInteractor.kt:29]: Just confirming, who will listen to event updates?

Comment (yshrsmz) [reply bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventInteractor.kt:29]: https://gist.github.com/yshrsmz/3690b1085dfe7edd1ec2218d9880bb38

EventForegroundScheduler will use this listener to send events when the cache exceeds the threshold and reschedule itself.

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Comment by yshrsmz** [comment]
Thanks!

**Pull request merged**